### PR TITLE
docs: mention what Tidelift and tidelift-me-up do

### LINKED ISSUE #18: 📝 Documentation: Mention what Tidelift does
### Bug Report Checklist

- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/tidelift-me-up-site/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

[Tidelift](https://tidelift.com) is a great thing but not everyone knows what it does. The docs here should mention it's a service that lets companies put money into a fund that gets distributed amongst their open source dependencies. Those dependencies need to show they do some good practices around maintenance & security in order to receive funds. It's a really good idea IMO - win win!

I'm wondering if I should put a (?) infotip/popover on the site too..

### Additional Info

Thanks @matthew-dean for pointing out that this is never mentioned anywhere!

Sibling issue to https://github.com/JoshuaKGoldberg/tidelift-me-up/issues/47.

### PR DESCRIPTION
**Pull request by maggienegm**
## PR Checklist

- [x] Addresses an existing open issue: fixes #18 
- [x] That issue was marked as [`status: accepting prs`](https://github.com/JoshuaKGoldberg/tidelift-me-up-site/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/JoshuaKGoldberg/tidelift-me-up-site/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

A description of Tidelift and tidelift-me-up was added to the `README.md`.

✨

### CODE REVIEW

**Review by JoshuaKGoldberg — APPROVED** [review]
💯 fantastic first repo contribution, thank you!

Cartoon french chef with making the chef's kiss gesture. Caption: "PERFECTO!"

File: README.md [hunk 1]
```diff
@@ -18,6 +18,16 @@
 	<img alt="TypeScript: Strict" src="https://img.shields.io/badge/typescript-strict-21bb42.svg" />
 </p>
 
+## What?
+
+[Tidelift](https://tidelift.com) is a service for companies to sponsor their open source dependencies.
+Through Tidelift, companies contribute to a fund that is distributed monthly to maintainers of eligible open-source packages in a process called "lifting."
```
Comment (JoshuaKGoldberg) [inline README.md]: Oh almost missed this:

```suggestion
Through Tidelift, companies contribute to a fund that is distributed monthly to maintainers of eligible open-source packages in a process called "lifting".
```

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Wait, sorry, _one_ little nitpick to request changes on 😄.

I recognize most English language style guides put punctuation inside quotes. But I disagree with that and especially in programming contexts think it's weird and inaccurate.

https://www.youtube.com/watch?v=UmZTzr0f5dw

**Comment by maggienegm** [comment]
No worries! I respect the council's decision 🫡 lol

**Review by JoshuaKGoldberg — APPROVED** [review]
🚀 thanks!

**Comment by JoshuaKGoldberg** [comment]
@all-contributors please add @maggienegm for doc.

> 🤖 Beep boop! This comment was added automatically by [all-contributors-auto-action](https://github.com/marketplace/actions/all-contributors-auto-action).
> Not all contributions can be detected from Git & GitHub alone. Please comment any missing contribution types this bot missed.
> ...and of course, thank you for contributing! 💙

**Pull request merged**